packages/framework/tests/Unit/Component/Money/MoneyTest.php

```diff
@@ -0,0 +1,582 @@
+    }
+
+    public function testMultiplyIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->multiply('2');
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider multiplyProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testMultiply(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->multiply($b);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function multiplyProvider(): Iterator
+    {
+        yield ['2', '1', '2'];
+        yield ['12.15', '34.965', '424.82475'];
+        yield ['10', '-2', '-20'];
+        yield ['1', '0.01', '0.01'];
+        yield ['0.5', '0.5', '0.25'];
+        yield ['1.525', '0.475', '0.724375'];
+        yield ['1.00', '1.000', '1.00000'];
+    }
+
+    public function testDivideIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->divide('2', 1);
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider divideProvider
+     * @param string $a
+     * @param string $b
+     * @param int $scale
+     * @param string $expectedResult
+     */
+    public function testDivide(string $a, string $b, int $scale, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+
+        $result = $moneyA->divide($b, $scale);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function divideProvider(): Iterator
+    {
+        yield ['1', '2', 0, '1'];
+        yield ['1', '2', 1, '0.5'];
+        yield ['1', '2', 2, '0.50'];
+        yield ['1', '3', 3, '0.333'];
+        yield ['2', '3', 3, '0.667'];
+        yield ['1000', '3', 3, '333.333'];
+        yield ['3.33', '3', 2, '1.11'];
+        yield ['0.1', '0.1', 0, '1'];
+    }
+
+    public function testModIsImmutable(): void
+    {
+        $money = Money::fromString('1');
+
+        $money->mod(Money::fromString('1'));
+
+        $this->assertEquals('1', $money->toString());
+    }
+
+    /**
+     * @dataProvider modProvider
+     * @param string $a
+     * @param string $b
+     * @param string $expectedResult
+     */
+    public function testMod(string $a, string $b, string $expectedResult): void
+    {
+        $moneyA = Money::fromString($a);
+        $moneyB = Money::fromString($b);
+
+        $result = $moneyA->mod($moneyB);
+
+        $this->assertEquals($expectedResult, $result->toString());
+    }
+
+    /**
+     * @return \Iterator
+     */
+    public function modProvider(): Iterator
+    {
+        yield ['100', '100', '0'];
+        yield ['120', '100', '20'];
+        yield ['530', '100', '30'];
+        yield ['-20', '100', '80'];
```

Comment:
After seeing this result I would prefer to remove `mod` operation from our `Money` because this is behavior of numbers, not money

Reply:
This might still be useful for example when rounding to the nearest fifty-cent. I would feel much more comfortable with using modulo in such case than multiplying, rounding and dividing the value.

Reply:
If we would have need rounding to the nearest 50-cent, then this rounding has to be part of the `Money` interface. So I would use `roundTo50` method, not `mod`.

So still, the `mod` operation is numbers behavior, not money behavior.

Reply:
It seems reasonable to remove `mod`. We can use some uglier method for the rounding to fifties...